### PR TITLE
[MySQL] `az mysql flexible-server export create`: Add Export Backup CLI implementation

### PR DESCRIPTION
**Pull request by nakulms**
**Related command**
az mysql flexible-server export create --% -g testgroup -n testsvr -b testbackup -u destsasuri

**Description**
We have implemented Export backup feature for MySQL flexible server. This code change is to enable customer access that feature through CLI

**Testing Guide**
az mysql flexible-server export create --% -g testgroup -n testsvr -b testbackup -u destsasuri
-g: Resource Group name
-n: Server name
-b: Backup name
-u: sas uri for destination container

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
MySQL

File: src/azure-cli/azure/cli/command_modules/mysql/_help.py [hunk 1]
```diff
@@ -754,3 +754,11 @@
   - name: Resets GTID '3E11FA47-71CA-11E1-9E33-C80AA9429562:23' on server 'testsvr'.
     text: az mysql flexible-server gtid reset -g testgroup -s testsvr --gtid-set 3E11FA47-71CA-11E1-9E33-C80AA9429562:23
 """
+
+helps['mysql flexible-server export create'] = """
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/_help.py:758]: Why don't you call it `mysql flexible-server export-backup create`?

I noticed that the client name and your description are all about export backup, not export

**Review by evelyn-ys — COMMENTED** [review]
Please add test for new command

File: src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py [hunk 2]
```diff
@@ -2169,3 +2171,83 @@ def _test_identity_aad_admin_mgmt(self, database_engine, resource_group, passwor
...
+
+    def _test_flexible_server_export_create_mgmt(self, database_engine, resource_group, server):     
+        account_name = self.create_random_name(STORAGE_ACCOUNT_PREFIX, STORAGE_ACCOUNT_NAME_MAX_LENGTH)
+        container_name = "mycontainer"
+        backup_name = "testexport"
+        location = DEFAULT_LOCATION
+
+        #create storgae account that will be used for export
+        self.cmd('storage account create --name {} --resource-group {} -l {} --sku Standard_LRS --output none'.format(account_name, resource_group, location))
+        target_account_info = self.get_account_info(resource_group, account_name)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py:2220]: We have `StorageAccountPreparer()` for creating a storage account for test. You don't need to imlement `get_account_info` by yourself

File: src/azure-cli/azure/cli/command_modules/mysql/_help.py (hunk 1 above)
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/_help.py:758]: Please add help for `mysql flexible-server export` command group too.

File: src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py (hunk 2 above)
Comment (nakulms) [reply src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py]: Done

File: src/azure-cli/azure/cli/command_modules/mysql/_help.py (hunk 1 above)
Comment (nakulms) [reply src/azure-cli/azure/cli/command_modules/mysql/_help.py:763]: Done

File: src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py [hunk 3]
```diff
@@ -2169,3 +2171,77 @@ def _test_identity_aad_admin_mgmt(self, database_engine, resource_group, passwor
...
+        if self.get_current_profile() == '2017-03-09-profile':
+            template = 'storage account keys list -n {} -g {} --query "key1" -otsv'
+        else:
+            template = 'storage account keys list -n {} -g {} --query "[0].value" -otsv'
+
+        return self.cmd(template.format(name, group)).output
+
+    def get_account_info(self, group, name):
+        """Returns the storage account name and key in a tuple"""
+        return name, self.get_account_key(group, name)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py:2193]:
```suggestion
```
The default profile is always `latest`. And you don't need to get storage account info by yourself, `StorageAccountPreparer()` already support returning account info

File: src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py [hunk 4]
```diff
@@ -2169,3 +2171,77 @@ def _test_identity_aad_admin_mgmt(self, database_engine, resource_group, passwor
...
+    @ServerPreparer(engine_type='mysql', location=DEFAULT_LOCATION)
+    @StorageAccountPreparer(location=DEFAULT_LOCATION)
+
+    def test_mysql_export(self, resource_group, server, storage_account):
+        self._test_flexible_server_export_create_mgmt('mysql', resource_group, server, storage_account)
+
+    def _test_flexible_server_export_create_mgmt(self, database_engine, resource_group, server, storage_account):     
+        backup_name = "testexport"
+
+        target_account_info = self.get_account_info(resource_group, storage_account)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py:2216]:
```suggestion
    @AllowLargeResponse()
    @ResourceGroupPreparer(location=DEFAULT_LOCATION)
    @ServerPreparer(engine_type='mysql', location=DEFAULT_LOCATION)
    @StorageAccountPreparer(location=DEFAULT_LOCATION)

    def test_mysql_export(self, resource_group, server, storage_account_info):
        self._test_flexible_server_export_create_mgmt('mysql', resource_group, server, storage_account_info)

    def _test_flexible_server_export_create_mgmt(self, database_engine, resource_group, server, storage_account_info):     
        backup_name = "testexport"

        target_account_info = storage_account_info
```

Comment (nakulms) [reply src/azure-cli/azure/cli/command_modules/mysql/tests/latest/test_mysql_scenario.py]: As discussed, moving test case to different PR

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**